modules/openfast-library/src/FAST_Subs.f90

```diff
@@ -2112,6 +2136,15 @@ SUBROUTINE FAST_InitOutput( p_FAST, y_FAST, Init, ErrStat, ErrMsg )
    
    NumOuts   = SUM( y_FAST%numOuts )
 
+   ! If already allocated from previous call, deallocate?
+   IF (ALLOCATED(y_FAST%ChannelNames)) THEN
+      DEALLOCATE(y_FAST%ChannelNames)
+   ENDIF 
+
+   IF (ALLOCATED(y_FAST%ChannelUnits)) THEN
```

Comment:
We could pre-emptively deallocate before returning the first time.  Might be less hacky.